apps/studio/src/server/modules/user/user.service.ts

```diff
@@ -11,3 +21,144 @@ export const isUserDeleted = async (email: string) => {
 
   return user?.deletedAt ? true : false
 }
+
+export const validateEmailRoleCombination = ({
+  email,
+  role,
+}: {
+  email: string
+  role: ResourcePermission["role"]
+}) => {
+  if (!isGovEmail(email) && role === RoleType.Admin) {
+    throw new TRPCError({
+      code: "FORBIDDEN",
+      message:
+        "Non-gov.sg emails cannot be added as admin. Select another role.",
+    })
+  }
+}
+
+interface CreateUserProps {
+  email: User["email"]
+  name?: User["name"]
+  phone?: User["phone"]
+  role: ResourcePermission["role"]
+  siteId: ResourcePermission["siteId"]
+  trx?: SafeKysely // allows for transaction to be passed in from parent transaction
```

Comment:
[question] jst from the naming - should this be a `tx` rather than accepting `tx | db`?

Reply:
good point, updated in https://github.com/opengovsg/isomer/pull/1118/commits/65a31f7059d452dd6b86e1c9b76be96eb575c8fa!